test-programs/compressed-token-test/tests/test.rs

```diff
@@ -243,27 +248,424 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+            ),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            light_system_program::errors::SystemProgramError::InvalidNoopPubkey.into(),
+        )
+        .unwrap();
+    }
+    // 8. Invalid account compression authority.
+    {
+        let invalid_account_compression_authority = Keypair::new();
+        let accounts = light_compressed_token::accounts::MintToInstruction {
+            fee_payer: payer_1.pubkey(),
+            authority: payer_1.pubkey(),
+            cpi_authority_pda: get_cpi_authority_pda().0,
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: invalid_account_compression_authority.pubkey(),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            anchor_lang::error::ErrorCode::ConstraintSeeds.into(),
+        )
+        .unwrap();
+    }
+    // 9. Invalid Merkle tree.
+    {
+        let invalid_merkle_tree = Keypair::new();
+        let instruction = create_mint_to_instruction(
+            &payer_1.pubkey(),
+            &payer_1.pubkey(),
+            &mint_1,
+            &invalid_merkle_tree.pubkey(),
+            amounts.clone(),
+            recipients.clone(),
+        );
+        rpc.create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await
+            .unwrap_err(); // Anchor panics, there is no specific error to assert.
```

Comment:
---- test_mint_to_failing stdout ----
forester: En9a97stB3Ek2n6Ey3NJwCUJnmTzLMMEA5C69upGDuQP
thread 'test_mint_to_failing' panicked at test-programs/compressed-token-test/tests/test.rs:625:14:
called `Result::unwrap()` on an `Err` value: TransactionError(InstructionError(0, ProgramFailedToComplete))

Reply:
We can either assert the error above or add a better one.

Reply:
I don't see much difference between asserting the generic `ProgramFailedToComplete` error and just doing `unwrap_err`, but OK, I can do that. The problem is that anchor-spl panics, so instead of having any error code, we get `ProgramFailedToComplete`.

This is also something which we could make an upstream PR for, BTW.

Reply:
Trying to assert `ProgramFailedToComplete` is a PITA...

`InstructionError` doesn't provide `From`/`Into` implementation which makes it convertable to `u32`, so

```rust
        let result = rpc
            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
            .await;
        // Anchor panics, there is no specific error to assert.
        assert_rpc_error(result, 0, InstructionError::ProgramFailedToComplete.into());

```

fails with:

```
error[E0277]: the trait bound `u32: From<solana_sdk::instruction::InstructionError>` is not satisfied
   --> test-programs/compressed-token-test/tests/test.rs:632:79
    |
632 |         assert_rpc_error(result, 0, InstructionError::ProgramFailedToComplete.into());
    |                                                                               ^^^^ the trait `From<solana_sdk::instruction::InstructionError>` is not implemented for `u32`, which is required by `solana_sdk::instruction::InstructionError: Into<_>`
    |
    = help: the following other types implement trait `From<T>`:
              <u32 as From<AccountCompressionErrorCode>>
              <u32 as From<Char>>
              <u32 as From<Ipv4Addr>>
              <u32 as From<SystemProgramError>>
              <u32 as From<TransferSdkError>>
              <u32 as From<VerifierError>>
              <u32 as From<anchor_lang::error::ErrorCode>>
              <u32 as From<bool>>
            and 23 others
    = note: required for `solana_sdk::instruction::InstructionError` to implement `Into<u32>
```

When trying to coerce it with `as u32` (which worked for other errors!):

```rust
        let result = rpc
            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
            .await;
        // Anchor panics, there is no specific error to assert.
        assert_rpc_error(result, 0, InstructionError::ProgramFailedToComplete as u32);
```

it complains with:

```
error[E0605]: non-primitive cast: `solana_sdk::instruction::InstructionError` as `u32`
   --> test-programs/compressed-token-test/tests/test.rs:632:37
    |
632 |         assert_rpc_error(result, 0, InstructionError::ProgramFailedToComplete as u32);
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can be used to convert enum types to numeric types only if the enum type is unit-only or field-less
```

because it's a data-carrying enum (some other variants of `InstructionError` are carrying other types).

For now I will just leave it as it is, sorry. Lesson for everyone to not panic.